models/settings_helpers.go

```diff
@@ -186,6 +193,13 @@ func ValidateSettings(params *ChangeSettingsParams) error {
 		}
 	}
 
+	if params.DisableSTT && params.EnableSTT {
+		return fmt.Errorf("enable STT and disable STT cannot be both true")
+	}
+	if params.EnableSTT && !params.EnableTelemetry {
```

Comment:
settings via API can be passed independently, So, params.EnableTelemetry may be false even if it was enabled in previous request. Please don't use `!` for validation